### DISCUSSION TITLE
Functions in example notebooks

### DISCUSSION THREAD

**Discussion by misi9170** [NREL/flasc#167] [General]
Many of the notebooks in the examples contain functions. I believe that many users, including ourselves, are simply copying the examples so that we can use the functions. Should these functions actually be part of the source code? That would:
1. Mean that they are available if flasc is installed via PyPI
2. Clean up the examples and make them easier to follow

The counterargument, I suppose, is that if the functions defined in the examples are too tied to the SCADA record, we shouldn't generalize them.

A good example of this is in examples_artificial_data/01_raw_data_processing/01_northing_calibration.ipynb. It feels like the following functions could all be moved to the source code:
- `get_bias_for_single_turbine()`
- `estimate_bias_with_reference_wd()`
- `apply_bias_corrections()`
- `filter_for_faults_in_impacting_turbines()`

**Comment by paulf81** [comment]
This strikes me as a good idea, @ejsimley if you agree I say we add this is as an issue.  We could then further discuss if this is a v2 target or future work

**Reply by ejsimley** [reply]
I agree, I haven't checked all the functions in the examples, but at least the ones @misi9170 listed could be generalized pretty easily and I feel would get used a lot outside of the examples.

**Comment by Bartdoekemeijer** [comment]
Agreed!
